packages/enzyme-adapter-react-renderer/src/ReactMountWrapper.js

```diff
@@ -0,0 +1,357 @@
+   * @returns {ReactWrapper}
+   */
+  setState(state, callback = noop) {
+    if (this.isRoot()) {
+      throw new Error('ReactWrapper::setState() can only be called on the root');
+    }
+    if (typeof callback !== 'function') {
+      throw new TypeError('ReactWrapper::setState() expects a function as its second argument');
+    }
+    this.instance().setState(state, () => {
+      callback();
+    });
+    return this;
+  }
+
+  /**
+   * Utility method that throws an error if the current instance has a length other than one.
+   * This is primarily used to enforce that certain methods are only run on a wrapper when it is
+   * wrapping a single node.
+   *
+   * @param {Function} fn
+   * @returns {*}
+   */
+  single(name, fn) {
+    const fnName = typeof name === 'string' ? name : 'unknown';
+    const callback = typeof fn === 'function' ? fn : name;
+    if (this.length !== 1) {
+      throw new Error(`Method “${fnName}” is only meant to be run on a single node. ${
+        this.length
+      } found instead.`);
+    }
+    return callback.call(this, this.instances[0]);
+  }
+
+  /**
+   * Returns the state hash for the root node of the wrapper. Optionally pass in a prop name and it
+   * will return just that value.
+   *
+   * NOTE: can only be called on a wrapper of a single node.
+   *
+   * @param {String} name (optional)
+   * @returns {*}
+   */
+  state(name) {
+    if (this.isRoot()) {
+      throw new Error('ReactWrapper::state() can only be called on the root');
+    }
+    const _state = this.single('state', instance => instance.instance.state);
+    if (typeof name !== 'undefined') {
+      return _state[name];
+    }
+    return _state;
+  }
+
+  /**
+   * Returns a string of the rendered text of the current render tree.  This function should be
+   * looked at with skepticism if being used to test what the actual HTML output of the component
+   * will be. If that is what you would like to test, use enzyme's `render` function instead.
+   *
+   * NOTE: can only be called on a wrapper of a single node.
+   *
+   * @returns {String}
+   */
+  text() {
+    return this.single(
+      'text',
+      instance => (instance.instance ? ReactDOM.findDOMNode(instance.instance).textContent : ''),
+    );
+  }
+
+  /**
+   * Returns the type of the root node of this wrapper. If it's a composite component, this will be
+   * the component constructor. If it's native DOM node, it will be a string.
+   *
+   * @returns {String|Function}
+   */
+  type() {
+    return this.single('type', instance => instance.type);
+  }
+}
+
+const createWrapper = (rootNode, passedOptions = {}) => {
+  const adapter = new ReactTestRendererAdapter();
+  const renderer = adapter.createMountRenderer(passedOptions);
+  const rootWrapper = renderer.render(rootNode, passedOptions.context);
+  const rootInstance = new ReactTestInstance(rootWrapper._reactInternalFiber);
+  return new ReactMountWrapper(rootInstance.children, rootWrapper, rootNode);
```

Comment:
This is the codepath that glues together the adapter and the wrapper. I create a new instance of `ReactTestInstance` and pass that to my wrapper. This is the functionality that we need to be exposed from `react-test-renderer`. Ideally, what is exposed should help me avoid having to reference `rootWrapper._reactInternalFiber`